nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-shell-authorizer/src/main/java/org/apache/nifi/authorization/NssShellCommands.java

```diff
@@ -85,6 +85,6 @@ public String getGroupById(String groupId) {
      * @return Shell command string that will exit normally (0) on a suitable system.
      */
     public String getSystemCheck() {
-        return "getent passwd";
+        return "getent passwd | cut -f 1,3,4 -d ':'";
```

Comment:
I'm not certain the system check needs a pipe and a cut, we're using that to determine if the host can provide `getent` and not any of the output.

Reply:
The problem is the result set can be really really large and the cut reduces the amount of info returned. In my test environment, without the cut, it was actually taking longer than the 30 second timeout on the Shell runner.

I was actually wondering if we do something that would be quicker, like can we assume every system with NSS has a user named rood and call "getent passwd root" to just look up one user instead of thousands?

Reply:
Maybe `which getent` is sufficient?  The only required behavior is for the check command to return non-zero.

Reply:
Let me try that quickly and if it works then I'll update the PR with that, if not I will leave it as is.

Reply:
Ok I tested it with `getent --version` and that worked so that should be a quicker alternative.